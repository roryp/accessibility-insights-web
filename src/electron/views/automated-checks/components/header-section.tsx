// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';
import * as styles from './header-section.scss';

export const HeaderSection = NamedFC('HeaderSection', () => {
    return (
        <div className={styles.headerSection}>
            <h1 className={styles.title}>Automated checks</h1>
            <p className={styles.subtitle}>
                Automated checks can detect some common accessibility problems such as missing or
                invalid properties. But most accessibility problems can only be discovered through
                manual testing.
            </p>
        </div>
    );
});
