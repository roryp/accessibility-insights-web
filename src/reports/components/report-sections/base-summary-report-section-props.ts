// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { ScanMetadata } from 'common/types/store-data/unified-data-interface';

export type BaseSummaryReportSectionProps = {
    toUtcString: (date: Date) => string;
    secondsToTimeString: (seconds: number) => string;
    getCollapsibleScript: () => string;
    scanMetadata: ScanMetadata;
};
