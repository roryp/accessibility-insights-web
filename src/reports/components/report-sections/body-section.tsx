// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import * as React from 'react';

export const BodySection = NamedFC('BodySection', ({ children }) => {
    return <body>{children}</body>;
});
