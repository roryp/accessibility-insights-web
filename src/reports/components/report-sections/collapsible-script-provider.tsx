// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

// Note: the source of this function's body is stringified and injected into the report.
//
// The use of function() {} syntax over arrow functions is important for IE compat (see #1875).
//
// The "istanbul ignore next" excludes the function from code coverage to prevent code cov from
// injecting functions that interfere with eval in the unit tests.
//
/* istanbul ignore next */
export const addEventListenerForCollapsibleSection = function (doc: Document): void {
    const collapsibles = doc.getElementsByClassName('collapsible-container');

    for (let index = 0; index < collapsibles.length; index++) {
        const container = collapsibles.item(index);
        const button = container.querySelector('.collapsible-control');
        button.addEventListener('click', function (): void {
            const content = button.parentElement.nextElementSibling as HTMLElement;
            const wasExpandedBefore =
                button.getAttribute('aria-expanded') === 'false' ? false : true;
            const isExpandedAfter = !wasExpandedBefore;

            button.setAttribute('aria-expanded', isExpandedAfter + '');
            content.setAttribute('aria-hidden', !isExpandedAfter + '');

            if (isExpandedAfter) {
                container.classList.remove('collapsed');
            } else {
                container.classList.add('collapsed');
            }
        });
    }
};

export const getAddListenerForCollapsibleSection = (code: string | Function): string =>
    `(${String(code)})(document)`;

export const getDefaultAddListenerForCollapsibleSection = (): string =>
    getAddListenerForCollapsibleSection(addEventListenerForCollapsibleSection);
