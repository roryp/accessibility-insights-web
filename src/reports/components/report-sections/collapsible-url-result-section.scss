// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
.url-result-section {
    padding-bottom: 31px;

    :global {
        .title-container[aria-level='3'] {
            .collapsible-control::before {
                position: relative;
                bottom: 2px;
                margin-right: 14px;
            }
        }

        .collapsible-content {
            margin-top: 19px;
        }
    }
}
