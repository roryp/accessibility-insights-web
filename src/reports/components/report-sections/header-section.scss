// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
@import '../../../common/styles/colors.scss';
@import '../../../common/styles/common.scss';
@import '../../../common/styles/fonts.scss';

.report-header-bar {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-wrap: nowrap;

    background-color: $ada-brand-color;
    height: $detailsViewHeaderBarHeight;
    width: 100%;

    :global(.header-icon) {
        flex-shrink: 0;
        margin-left: 16px;
        height: 22px;
        width: 22px;
    }

    .header-text {
        margin-left: 8px;

        color: $header-bar-title-color;

        flex-shrink: 1;
        @include ellipsedText();
        @include text-style-header-s;
    }
}

.report-header-command-bar {
    height: 40px;
    width: 100%;
    display: flex;
    justify-content: end;
    min-height: fit-content;
    align-items: center;
    background-color: $neutral-0;
    box-shadow: 0px 1px 0px rgba(0, 0, 0, 0.08);

    .target-page {
        @include ellipsedText();

        margin: 0px 0px 0px 16px;
        display: inherit;

        color: $primary-text;
        font-family: $fontFamily;
        font-size: 14px;
        line-height: 20px;
        font-weight: normal;

        a {
            @include ellipsedText();
        }
    }
}
