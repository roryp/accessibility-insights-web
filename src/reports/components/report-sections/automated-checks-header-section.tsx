// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { ScanMetadata } from 'common/types/store-data/unified-data-interface';
import * as React from 'react';
import { HeaderSection } from 'reports/components/report-sections/header-section';

export interface AutomatedChecksHeaderSectionProps {
    scanMetadata: ScanMetadata;
}

export const AutomatedChecksHeaderSection = NamedFC<AutomatedChecksHeaderSectionProps>(
    'AutomatedChecksHeaderSection',
    ({ scanMetadata }) => {
        return <HeaderSection targetAppInfo={scanMetadata.targetAppInfo} />;
    },
);
