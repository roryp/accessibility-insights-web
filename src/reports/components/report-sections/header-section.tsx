// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { NamedFC } from 'common/react/named-fc';
import { TargetAppData } from 'common/types/store-data/unified-data-interface';
import { productName } from 'content/strings/application';
import { BrandWhite } from 'icons/brand/white/brand-white';
import * as React from 'react';
import { NewTabLinkWithConfirmationDialog } from 'reports/components/new-tab-link-confirmation-dialog';
import * as styles from './header-section.scss';

export interface HeaderSectionProps {
    targetAppInfo: TargetAppData;
}

export const HeaderSection = NamedFC<HeaderSectionProps>('HeaderSection', ({ targetAppInfo }) => {
    return (
        <header>
            <div className={styles.reportHeaderBar}>
                <BrandWhite />
                <div className={styles.headerText}>{productName}</div>
            </div>
            <div className={styles.reportHeaderCommandBar}>
                <div className={styles.targetPage}>
                    Target page:&nbsp;
                    <NewTabLinkWithConfirmationDialog
                        href={targetAppInfo.url}
                        title={targetAppInfo.name}
                    >
                        {targetAppInfo.name}
                    </NewTabLinkWithConfirmationDialog>
                </div>
            </div>
        </header>
    );
});
