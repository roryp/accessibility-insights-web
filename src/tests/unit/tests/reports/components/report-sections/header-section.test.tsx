// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { shallow } from 'enzyme';
import * as React from 'react';
import { HeaderSection } from 'reports/components/report-sections/header-section';

describe('HeaderSection', () => {
    it('renders', () => {
        const targetAppInfo = {
            name: 'page-title',
            url: 'url://page',
        };
        const wrapper = shallow(<HeaderSection targetAppInfo={targetAppInfo} />);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
