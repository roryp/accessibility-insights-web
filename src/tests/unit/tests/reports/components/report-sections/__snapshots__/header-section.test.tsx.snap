// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSection renders 1`] = `
<header>
  <div
    className="reportHeaderBar"
  >
    <BrandWhite />
    <div
      className="headerText"
    >
      Accessibility Insights for Web
    </div>
  </div>
  <div
    className="reportHeaderCommandBar"
  >
    <div
      className="targetPage"
    >
      Target page:Â 
      <NewTabLinkWithConfirmationDialog
        href="url://page"
        title="page-title"
      >
        page-title
      </NewTabLinkWithConfirmationDialog>
    </div>
  </div>
</header>
`;
